### ISSUE TITLE
edit-pr-comment doesn't behave as expected

### ISSUE THREAD

**Issue by blampe** [pulumi/actions#519]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Our PRs using `pulumi/actions@v3` for previews have a lot of Pulumi noise despite explicitly using `edit-pr-comment: true`. 

After digging a little, it seems this logic captures the comment editing logic:

https://github.com/pulumi/actions/blob/26e4f94ae425f2b5b098e1398073ef35e7bf02fa/src/libs/pr.ts

```
  const heading = `#### :tropical_drink: \`${command}\` on ${stackName}`;
  ...
    if (editCommentOnPr) {
      const { data: comments } = await octokit.rest.issues.listComments({
        ...repo,
        issue_number: payload.pull_request.number,
      });
      const comment = comments.find((comment) =>
        comment.body.startsWith(heading),
      );
  ...
```

however after checking some of the noisy comments we're seeing, it seems they use a different header which doesn't get picked up by this `startsWith(heading)` check. Here's an example:

```
:tropical_drink: The **[Update (preview)](https://app.pulumi.com/...)** for stack **[stack/name](https://app.pulumi.com/...)** was successful.
```

Note they differ at least on "on ${stack}" vs. "for stack...".

Edit: I would expect to see one comment on the PR which has been edited to reflect the latest status of the update/preview result. Instead, we see one comment for every time the operation has run.

**Comment by blampe** [comment]
I narrowed down part of the confusion after realizing my action actually had `comment-on-pr: false`!

There are actually 2 flavors of comments, plaintext/console ones coming from this action and richer ones (colors, URLs...) coming from the GitHub app https://github.com/apps/pulumi

I had completely forgotten we had this app installed, which explains why we kept getting duplicated comments despite the action's settings.

I re-ran the action with `comment-on-pr` and `edit-pr-comment` both `true` and it appears to work as expected, so it seems the real issue is that the Github app doesn't share the same edit-in-place behavior the action defaults to. Perhaps this is a dupe of https://github.com/pulumi/service-requests/issues/30.
